osgar/tools/lidarview.py

```diff
@@ -416,14 +416,15 @@ def _step(self, direction):
                 arr = deserialize(data)
                 assert len(arr) == 3
                 self.pose = (arr[0]/1000.0, arr[1]/1000.0, math.radians(arr[2]/100.0))
+            if stream_id in [self.pose2d_id, self.pose3d_id]:
                 x, y, heading = self.pose
                 self.pose = (x * math.cos(g_rotation_offset_rad) - y * math.sin(g_rotation_offset_rad),
                              x * math.sin(g_rotation_offset_rad) + y * math.cos(g_rotation_offset_rad),
                              heading + g_rotation_offset_rad)
-                if self.lidar_id is None and self.camera_id is None:
-                    keyframe = self.keyframe
-                    self.keyframe = False
-                    return timestamp, self.frame, self.pose, self.pose3d, self.scan, self.scan2, self.image, self.image2, self.bbox, self.joint, keyframe, self.title, False
+            if self.lidar_id is None and self.camera_id is None:
```

Comment:
this if section not lidar and not camera is really INSIDE the processing of pose (future pose3d) and it is ment to define keyframe (lidar is primary, camera secondary and pose is tertiary)